src/test/java/io/cdap/plugin/gcp/dataplex/source/DataplexInputFormatProviderTest.java

```diff
@@ -0,0 +1,105 @@
+import io.cdap.plugin.gcp.dataplex.common.util.DataplexConstants;
+import io.cdap.plugin.gcp.dataplex.common.util.DataplexUtil;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.mapreduce.InputFormat;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Spliterator;
+import java.util.stream.StreamSupport;
+
+import static org.powermock.api.mockito.PowerMockito.mock;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ConfigurationUtils.class, GCPUtils.class, GoogleCredentials.class, DataplexUtil.class})
+public class DataplexInputFormatProviderTest {
+
+  @Mock
+  JobContext mockContext;
+
+  @Mock
+  private Configuration configuration;
+
+  @Mock
+  private InputFormat inputFormat;
+
+  @Mock
+  private InputSplit inputSplit;
+
+  @Mock
+  private TaskAttemptContext context;
+
+  @Mock
+  private ConfigurationUtils configurationUtils;
+
+  @Test
+  public void testGetInputFormatClassName() throws Exception {
+    Configuration configuration = mock(Configuration.class);
+    PowerMockito.mockStatic(StreamSupport.class);
+    String entityType = StorageSystem.CLOUD_STORAGE.toString();
+    when(configuration.get(DataplexConstants.DATAPLEX_ENTITY_TYPE)).thenReturn(entityType);
+    PowerMockito.mockStatic(ConfigurationUtils.class);
+    DataplexInputFormatProvider.DataplexInputFormat dataInputFormat = new
+      DataplexInputFormatProvider.DataplexInputFormat();
+    Map<String, String> map = new HashMap<>();
+    map.put("price", "100");
+    Mockito.when(ConfigurationUtils.getNonDefaultConfigurations(configuration)).thenReturn(map);
+    DataplexInputFormatProvider dataplexInputFormatProvider = new DataplexInputFormatProvider(configuration);
+    Assert.assertThrows(ClassCastException.class, () -> {
+      dataInputFormat.createRecordReader(inputSplit, context);
+    });
+    Assert.assertEquals("io.cdap.plugin.gcp.dataplex.source.DataplexInputFormatProvider" +
+      "$DataplexInputFormat", dataplexInputFormatProvider.getInputFormatClassName());
+  }
+
+  @Test
+  public void testGetInputFormatDataplexClassNameWithBigquery() throws Exception {
+    Configuration configuration = mock(Configuration.class);
+    Map<String, String> inputFormatConfiguration = new HashMap<>();
+    DataplexInputFormatProvider.DataplexInputFormat dataplexInputFormat = new
+      DataplexInputFormatProvider.DataplexInputFormat();
+    inputFormatConfiguration.put("key", "value");
```

Comment:
Are we making use of `inputFormatConfiguration` after putting values in it?

Reply:
Yes it was unused so removed it.